src/librustdoc/html/render/print_item.rs

```diff
@@ -1420,30 +1420,36 @@ fn item_macro(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean:
     write!(w, "{}", document(cx, it, None, HeadingOffset::H2))
 }
 
-fn item_proc_macro(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, m: &clean::ProcMacro) {
-    wrap_item(w, |w| {
+fn item_proc_macro(
+    w: &mut impl fmt::Write,
+    cx: &mut Context<'_>,
+    it: &clean::Item,
+    m: &clean::ProcMacro,
+) {
+    let mut buffer = Buffer::new();
```

Comment:
*Just passing by on some changes happening around :)*

Wondering, does the usage of `Buffer` here costs us some extra allocations? I believe we can use `display_fn()` to reduce that (?)

CMIIW @GuillaumeGomez

Reply:
This are some of my initial contributions to the compiler : ) 
I was thinking  might not be a straightforward replacement in this case, as you'd need to handle the different `MacroKind `cases inside the provided function, which could make the code a bit more complex. Also, it would require changing `wrap_item function`, since we are passing a closure that modifies a buffer.

Reply:
Well, simplest answer: let's run perf check on this PR. :)